firefox_ui_tests/functional/locationbar/test_go_button.py

```diff
@@ -0,0 +1,31 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from firefox_ui_harness.testcase import FirefoxTestCase
+
+from marionette_driver import By, Wait
+
+
+class TestGoButton(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+        self.url = self.marionette.absolute_url('layout/mozilla.html')
+
+    def test_gobutton(self):
+        # no input, go button should not display
+        go_button = self.browser.locationbar.go_button
+        self.assertEqual(go_button.element.get_attribute('state'), 'disabled')
```

Comment:
The state will not be clear because you don't have the initial page load anymore. Keep in mind that you don't know how the former test left the browser.